### ISSUE TITLE
S-Lora for cost optimization

### ISSUE THREAD

**Issue by flozi00** [huggingface/text-generation-inference#1250]
### Feature request

https://github.com/S-LoRA/S-LoRA

### Motivation

Integrate S-Lora into tgi could be a huge cost optimization step for the hosted huggingface tgi endpoints
Since its possible to get an lora out of two full models it would be a good chance to lower the costs for the hosted endpoints included in pro plan on hf hub.

Example:
Mistral as base model, zephyr as finetune --> lora for the zephyr model
https://github.com/uukuguy/multi_loras

Both model versions could be hosted in single tgi and would be a huge step for load balancing

### Your contribution

I have not enough free time over the next weeks

**Comment by flozi00** [comment]
https://github.com/predibase/lorax

Lorax is open source now
It's even based on tgi
@OlivierDehaene @Narsil

**Comment by malhajar17** [comment]
@Narsil I would love to help out in this integration

**Comment by Narsil** [comment]
Definitely something we discussed internally, even before lorax was out.

Currently not super high in our prio list, given that not many users are running **multiple** loras at the same time (at least from our understanding).

We have implemented something like that here for diffusers: https://github.com/huggingface/api-inference-community/blob/main/docker_images/diffusers/app/pipelines/text_to_image.py#L25

Definitely a great cost reduction overall since we're pooling resources for all SD 1.5 loras now.

(Implementation is simpler because diffusers is much more compute bound that memory bound like LLMs are.)
